frontend/app/src/components/StatusWidget/StatusWidget.tsx

```diff
@@ -136,237 +108,244 @@ const RUNNING_MAN_DISPLAY_DELAY_TIME_MS = 500
+        minimizePromptTimer.current.setTimeout(() => {
+          setPromptMinimized(true)
+        }, timeout)
+      }
     }
-  }
+  }, [])
+
+  const handleSessionEvent = useCallback(
+    (event: SessionEvent): void => {
+      if (event.type === "scriptChangedOnDisk") {
+        setScriptChangedOnDisk(true)
+        setPromptMinimized(false)
+        minimizePromptAfterTimeout(PROMPT_DISPLAY_INITIAL_TIMEOUT_MS)
+      }
+    },
+    [minimizePromptAfterTimeout]
+  )
+
+  const showRunningManAfterInitialDelay = useCallback(
+    (delay: number): void => {
+      if (showRunningManTimer.current !== null) {
+        showRunningManTimer.current.setTimeout(() => {
+          setShowRunningMan(true)
+        }, delay)
+      }
+    },
+    []
+  )
 
-  handleKeyDown = (keyName: string): void => {
-    // NOTE: 'r' is handled at the App Level
-    if (keyName === "a") {
-      this.handleAlwaysRerunClick()
-    }
-  }
+  const handleScroll = useCallback((): void => {
+    setStatusMinimized(shouldMinimize())
+  }, [])
 
-  /** Called by React on prop changes */
-  public static getDerivedStateFromProps(
-    props: Readonly<StatusWidgetProps>
-  ): Partial<State> | null {
-    // Reset transient event-related state when prop changes
-    // render that state irrelevant
-    if (props.scriptRunState === ScriptRunState.RUNNING) {
-      return { scriptChangedOnDisk: false, promptHovered: false }
-    }
+  const onAppPromptHover = useCallback((): void => {
+    setPromptHovered(true)
+  }, [])
 
-    return null
-  }
+  const onAppPromptUnhover = useCallback((): void => {
+    setPromptHovered(false)
+    setPromptMinimized(false)
+    minimizePromptAfterTimeout(PROMPT_DISPLAY_HOVER_TIMEOUT_MS)
+  }, [minimizePromptAfterTimeout])
 
-  public componentDidMount(): void {
-    this.sessionEventConn =
-      this.props.sessionEventDispatcher.onSessionEvent.connect(e =>
-        this.handleSessionEvent(e)
-      )
-    window.addEventListener("scroll", this.handleScroll)
-  }
+  const handleStopScriptClick = useCallback((): void => {
+    stopScript()
+  }, [stopScript])
 
-  public componentWillUnmount(): void {
-    if (this.sessionEventConn !== undefined) {
-      this.sessionEventConn.disconnect()
-      this.sessionEventConn = undefined
-    }
+  const handleRerunClick = useCallback((): void => {
+    rerunScript(false)
+  }, [rerunScript])
 
-    this.minimizePromptTimer.cancel()
-    this.showRunningManTimer.cancel()
+  const isNewYears = useCallback((): boolean => {
+    // Test if current date between 12/31 & 1/06
+    const currentDate = new Date()
+    const month = currentDate.getMonth()
+    const date = currentDate.getDate()
+    // Check if Dec 31st
+    if (month === 11 && date === 31) return true
+    // Check if Jan 1st through 6th
+    if (month === 0 && date <= 6) return true
+    return false
+  }, [])
 
-    window.removeEventListener("scroll", this.handleScroll)
+  function shouldMinimize(): boolean {
```

Comment:
Considering re-writing these functions as named functions using `useCallback` as well.

Reply:
**question:** How about an inline variable? `const shouldMinimize = window.scrollY > 32` is a pretty simple boolean calculation that doesn't require additional overhead of hooks

Reply:
I think this will need to remain a function so that `window.scrollY` will be checked dynamically on the scroll event. Point taken about `useCallback` though, I was thinking this hook should be used more often than it really should be.

Reply:
Oh, good point. 

As an alternative, it could also be re-written like this: 

```
const MINIMIZE_ON_SCROLL_Y = 32

...
const [statusMinimized, setStatusMinimized] = useState(window.scrollY > MINIMIZE_ON_SCROLL_Y)

...

 const handleScroll = useCallback((): void => {
    setStatusMinimized(window.scrollY > MINIMIZE_ON_SCROLL_Y)
  }, [])

```

The downside it is still some repeated logic that future developers will need to know to update in both places, but these two areas are linked together by the state variable so it seems intuitive to me.